src/libs/ReportUtils.js

```diff
@@ -4115,6 +4116,30 @@ function getIOUReportActionDisplayMessage(reportAction) {
     return displayMessage;
 }
 
+/**
+ * Return room channel log display message
+ * @param {Object} reportAction
+ * @returns {String}
+ */
+function getRoomChannelLogMemberMessage(reportAction) {
```

Comment:
Looks better! But I have a suggestion to increase the simplicity of the function. Let me know if you agree.

```js
/**
 * Return room channel log display message
 * 
 * @param {Object} reportAction
 * @returns {String}
 */
function getRoomChannelLogMemberMessage(reportAction) {
    const actionPerformed = reportAction.actionName === CONST.REPORT.ACTIONS.TYPE.ROOMCHANGELOG.INVITE_TO_ROOM ? 'invited' : 'removed';

    const mentions = _.map(reportAction.originalMessage.targetAccountIDs, (accountID) => {
        const personalDetail = lodashGet(allPersonalDetails, accountID);
        const displayNameOrLogin =
            LocalePhoneNumber.formatPhoneNumber(lodashGet(personalDetail, 'login', '')) || lodashGet(personalDetail, 'displayName', '') || Localize.translateLocal('common.hidden');
        return `@${displayNameOrLogin}`;
    });

    const lastMention = mentions.pop();

    if (mentions.length === 0) {
        return `${actionPerformed} ${lastMention}`;
    }

    return `${actionPerformed} ${mentions.join(', ')} and ${lastMention}`;
}
```